File: 1zadanie.go
package main

import (
	"fmt"
	"math"
)

// Кусочно-заданная функция
func f(x float64) float64 {

	// 1. Левая полуокружность
	if x >= -6.0 && x <= -4.0 {
		return -math.Sqrt(4.0 - (x+4.0)*(x+4.0))
	}

	// 2. y = cbrt(2x)
	if x > -4.0 && x <= 4.0 {
		return math.Cbrt(2.0 * x)
	}

	// 3. Правая полуокружность
	if x > 4.0 && x <= 6.0 {
		return math.Sqrt(4.0 - (x-4.0)*(x-4.0))
	}

	// Вне заданных участков графика
	return 0.0
}

func main() {
	var Xn, Xk, dx float64

	fmt.Print("Введите начальный X, конечный X, шаг dx: ")
	fmt.Scan(&Xn, &Xk, &dx)

	if dx <= 0 {
		fmt.Println("Ошибка: шаг должен быть положительным!")
		return
	}

	fmt.Println("\n------------------------------------------")
	fmt.Println("|     x    |             y              |")
	fmt.Println("------------------------------------------")

	for x := Xn; x <= Xk+1e-9; x += dx {
		y := f(x)

		fmt.Printf("| %8.4f | %24.4f |\n", x, y)
	}

	fmt.Println("------------------------------------------")
}

File: 1zadanie.cpp
#include <iostream>
#include <iomanip>
#include <cmath>


double f(double x) {

    // 1. Левая полуокружность
    if (x >= -6.0 && x <= -4.0) {
        return -sqrt(4.0 - (x + 4.0) * (x + 4.0));
    }

    // 2. y = cbrt(2x)
    else if (x > -4.0 && x <= 4.0) {
        return cbrt(2.0 * x);
    }

    // 3. Правая полуокружность
    else if (x > 4.0 && x <= 6.0) {
        return sqrt(4.0 - (x - 4.0) * (x - 4.0));
    }

    // Вне заданных участков графика
    else {
        return 0.0;
    }
}

int main() {
    double Xn, Xk, dx;

    std::cout << "Введите начальный X, конечный X, шаг dx: ";
    std::cin >> Xn >> Xk >> dx;

    // Проверка шага
    if (dx <= 0) {
        std::cout << "Ошибка: шаг должен быть положительным!" << std::endl;
        return 1;
    }

    std::cout << std::fixed << std::setprecision(4);

    std::cout << "\n-------------------------------\n";
    std::cout << "|     x        |       y      |\n";
    std::cout << "-------------------------------\n";

    for (double x = Xn; x <= Xk + 1e-9; x += dx) {

        if (x < -6.0 || x > 6.0) {
            std::cout << "| "
                    << std::setw(12) << x
                    << " | "
                    << std::setw(12) << "Not defined"
                    << " |\n";
        }
        else {
            double y = f(x);

            std::cout << "| "
                    << std::setw(12) << x
                    << " | "
                    << std::setw(12) << y
                    << " |\n";
        }
    }

    std::cout << "-------------------------------\n";

    return 0;
}